### PR TITLE
feat(carousel): add ability to swipe slides

### PR DESCRIPTION
**Pull request by wesleycho**
- Add ability to swipe carousel slides on mobile

This should address #1098

### CODE REVIEW

File: src/carousel/carousel.ts [hunk 1]
```diff
@@ -157,6 +157,14 @@ export class NgbCarousel implements AfterContentChecked,
     }
   }
 
+  swipeCycle(event) {
```
Comment (pkozlowski-opensource) [inline src/carousel/carousel.ts]: `event` -> `$event`

**Comment by pkozlowski-opensource** [comment]
Cool @wesleycho ! Could you please add test for this as well?

**Comment by wesleycho** [comment]
Good thing you asked for a test - turned out I put the swipe on the wrong element!

**Comment by wesleycho** [comment]
There is one unfortunate side effect of the addition of the swipe - it seems to require hammer.js to be loaded. How do you feel about this @pkozlowski-opensource ?

**Comment by wesleycho** [comment]
I just opened https://github.com/angular/angular/issues/16992 since I think this is important for Angular to support.

**Comment by Only1MrAnderson** [comment]
any traction on this?

**Comment by pkozlowski-opensource** [comment]
So I looked into the topic a bit and it seems like `hammer.js` is not actively supported any more. As such I wouldn't like to tie ng-bootstrap to it. We will need to find another approach in #1098

**Pull request closed without merging**